src/main/java/Main.java

```diff
@@ -1,8 +1,42 @@
-public class Main {
-
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.*;
 
+public class Main {
     public static void main(String[] args) {
+        try {
+            List<Integer> numbers = Main.readFile("liczby.txt");
+            System.out.println(numbers);
+            Map<Integer, Integer> occurrences = findOccurrences(numbers, "liczby.txt");
+            System.out.println(occurrences);
+        } catch (FileNotFoundException e) {
+            System.err.println("Nie udało się odczytać pliku.");
+        }
 
     }
 
+    private static Map<Integer, Integer> findOccurrences(List<Integer> numbers, String fileName) throws FileNotFoundException {
+        Map<Integer, Integer> occurrences = new HashMap<>();
+        Scanner scanner = new Scanner(new File(fileName));
+        int number = 0;
+        while (scanner.hasNextLine()) {
+            int line = Integer.parseInt(scanner.nextLine());
+            if (numbers.contains(line)) {
+                number++;
+                occurrences.put(line, number);
+            }
+        }
+        return occurrences;
+    }
```

Comment:
Ta metoda nie powinna drugi raz wczytywać z pliku. Powinna przejść po wszystkich elementach listy i wstawiać wartości do mapy. W pseudokodzie:
1. Dla każdego elementu listy:
2. Sprawdź, czy w mapie jest już taka liczba
3.1. Jeżeli nie, to wstaw wartość do mapy z liczbą wystąpień równą 1
3.2. Jeżeli tak, to pobierz liczbę wystąpień z mapy, zwiększ ją o 1 i ponownie zapisz do mapy

Możesz też obczaić metodę merge w mapach